src/tblib/__init__.py

```diff
@@ -88,3 +97,47 @@ def __tproxy_handler(self, operation, *args, **kwargs):
                 return getattr(self, args[0])
         else:
             return getattr(self, operation)(*args, **kwargs)
+
+    def to_dict(self):
+        """Convert a Traceback into a dictionary representation"""
+        if self.tb_next is None:
+            tb_next = None
+        else:
+            tb_next = self.tb_next.to_dict()
+
+        code = {
+            k: v
+            for k, v in self.tb_frame.f_code.__dict__.items()
+            if k.startswith('co_')
+        }
+        code['co_lnotab'] = code['co_lnotab'].decode('latin1')
+        frame = {
+            'f_globals': self.tb_frame.f_globals,
+            'f_code': code
+        }
+        return {
+            'tb_frame': frame,
+            'tb_lineno': self.tb_lineno,
+            'tb_next': tb_next
+        }
+
+    @classmethod
+    def from_dict(cls, dct):
+        if dct['tb_next']:
+            tb_next = cls.from_dict(dct['tb_next'])
+        else:
+            tb_next = None
+
+        frame = _AttrDict((
+            ('f_globals', dct['tb_frame']['f_globals']),
+            ('f_code', _AttrDict((k, v) for k, v in dct['tb_frame']['f_code'].items()))
+        ))
+        frame['f_code']['co_lnotab'] = frame['f_code']['co_lnotab'].encode('latin1')
```

Comment:
Actually, I just found out 'hex' doesn't work, it  requires that you have an even-length bytes to decode.

The tricky part about encoding co_lnotab is that you start with bytes, but json requires unicode. So you have to find an encoding that maps any arbitrary bytes to unicode code points, and then back without any changes. It would definitely be nice if it weren't required.